llpc/util/llpcElfWriter.cpp

```diff
@@ -1004,6 +1080,96 @@ void ElfWriter<Elf>::Reinitialize()
     m_map[NoteName] = m_noteSecIdx;
 }
 
+uint32_t GetResourceOffset(uint32_t descSet, uint32_t binding, const ResourceMappingNode* userResources, uint32_t
+    resourceCount);
+
+// =====================================================================================================================
+// Fixup relocations in the ELF with actual values.
+template<class Elf>
+void FixUpRelocations(
+    ElfWriter<Elf> *pWriter,                        // [InOut] Writer to target ELF
+    const std::vector<ElfReloc>& relocations,       // Relocation entries
+    Context *pContext)                              // Pipeline compilation context
+{
+    for(uint32_t i = 0; i < relocations.size(); ++i)
+    {
+        ElfReloc reloc = relocations[i];
+
+        ElfSymbol sym;
+        pWriter->GetSymbol(reloc.symIdx, &sym);
+
+        if (strncmp(sym.pSymName, "doff_", 5) != 0)
+        {
+            continue;
+        }
+
+        size_t idx = 0;
+        uint32_t descSet = std::stoi(sym.pSymName+5, &idx);
+        uint32_t binding = std::stoi(sym.pSymName+6+idx);
+        uint32_t offset = 0;
+
+        auto pPipelineInfo = reinterpret_cast<const GraphicsPipelineBuildInfo*>(pContext->GetPipelineBuildInfo());
+
+        offset = GetResourceOffset(descSet, binding, pPipelineInfo->vs.pUserDataNodes, pPipelineInfo->vs
+        .userDataNodeCount);
+        if (offset == InvalidValue)
+        {
+            offset = GetResourceOffset(descSet, binding, pPipelineInfo->fs.pUserDataNodes, pPipelineInfo->fs
+                    .userDataNodeCount);
```

Comment:
If I don't take mistake, this offset is a relevant offset in second level of resource nodding map. Do you know how we deal with root offset?